### DISCUSSION TITLE
PV to azure disk mapping

### DISCUSSION THREAD

**Discussion by sarav-ks** [kubernetes-sigs/azuredisk-csi-driver#1775] [Q&A] [answered]
We are planning to use 'managed-csi' storage class to create PV , can you help clarify the below questions 

1) Does every pod create a new Azure Managed Disk ? I understand it is 'ReadWriteOnce' , hence a disk can be used only by a single pod at a time .
2) Lets say if we are spinning up 200 pods at the same time , will it create 200 Azure disks ?
3) Would it be possible to create a disk of size 500 GB and use the same disk with multiple pods which PV's of smaller size (example : 25GB , 100 GB , 75 GB, 300 GB) ?

Mainly i wanted to understand if the PV to Azure Disk mapping is "one to one" or "many to one"

**Comment by zohebk8s** [comment]
@sarav-ks It depends on your requirements.

If you want multiple pods to access a single storage volume, then explore more about Azure Files.

Azure Disk are mounted as ReadWriteOnce, they're only available to a single node. For storage volumes that can be accessed by pods on multiple nodes simultaneously, use Azure Files.

**Reply by zohebk8s** [reply, marked as answer]
https://learn.microsoft.com/en-us/azure/aks/concepts-storage
